src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -258,83 +193,34 @@ private IEnumerable<ITypeSymbol> InferTypeInArrowExpressionClause(ArrowExpressio
-                        // In this case, we need to detect that the 'new' token is on the left side of 
-                        // its parent assignment and on the right side of its grandparent assignment and
-                        // return the inferred type of the grandparent.
-                        if (parentAssignment.Parent is AssignmentExpressionSyntax)
-                        {
-                            var grandparentAssignment = (AssignmentExpressionSyntax)parentAssignment.Parent;
-                            if (grandparentAssignment.IsKind(SyntaxKind.SimpleAssignmentExpression) &&
-                                parentAssignment == grandparentAssignment.Right)
-                            {
-                                parent = grandparentAssignment;
-                                token = grandparentAssignment.OperatorToken;
-                            }
-                        }
-                        // case 2:
-                        //      bool b;
-                        //      Task task = new $$
-                        //      b = false;
-                        // The 'new' token is part of an assignment of the assignment to b,
-                        // but the user is really trying to assign to task.
-                        // In this case, we need to detect that the 'new' token is on the left side of 
-                        // its parent assignment and on the right side of its grandparent assignment and
-                        // return the inferred type of the grandparent.
-                        else if (parentAssignment.Parent is EqualsValueClauseSyntax)
-                        {
-                            var grandParentAssignment = (EqualsValueClauseSyntax)parentAssignment.Parent;
-                            if (parentAssignment == grandParentAssignment.Value)
-                            {
-                                parent = grandParentAssignment;
-                                token = grandParentAssignment.EqualsToken;
-                            }
-                        }
-                    }
-                }
+                var parent = token.Parent;
 
                 return parent.TypeSwitch(
                     (AnonymousObjectMemberDeclaratorSyntax memberDeclarator) => InferTypeInMemberDeclarator(memberDeclarator, token),
-                    (ArgumentSyntax argument) => InferTypeInArgument(argument, token),
                     (ArgumentListSyntax argument) => InferTypeInArgumentList(argument, token),
-                    (AttributeArgumentSyntax argument) => InferTypeInAttributeArgument(argument, token),
-                    (AttributeArgumentListSyntax attributeArgumentList) => InferTypeInAttributeArgumentList(attributeArgumentList, token),
-                    (CheckedExpressionSyntax checkedExpression) => InferTypes(checkedExpression),
+                    (ArgumentSyntax argument) => InferTypeInArgument(argument, token),
```

Comment:
I chose not to order these when I was in here because I was concerned that one of condition might need to be checked before another. Looking again, I'm more comfortable with it.